api/app/Models/PoolCandidate.php

```diff
@@ -536,6 +540,8 @@ public function scopeSkillsIntersectional(Builder $query, ?array $skills): Build
             return $query;
         }
 
+        $query = $this->addSkillCountSelect($query, $skills);
+
         // call the skillFilter off connected user
         $query->whereHas('user', function (Builder $userQuery) use ($skills) {
             User::scopeSkillsIntersectional($userQuery, $skills);
```

Comment:
There isn't much of a point to adding it to `scopeSkillsIntersectional` though is there? It will always return the number of skills selected as it is AND matching.

But it does make things consistent and easy to swap skill fields by having consistent typing/return value 😕

Reply:
That is a very good point 😆